generators/python-v2/ast/src/Reference.ts

```diff
@@ -43,7 +43,8 @@ export class Reference extends AstNode {
     }
 
     public write(writer: Writer): void {
-        writer.write(this.alias ?? this.name);
+        const nameOverride = writer.getRefNameOverride(this);
+        writer.write(nameOverride.name);
```

Comment:
Notably a `Reference` object now gets its name from the `Writer` during its own `write` method.